### ISSUE TITLE
Mesa 10.3.5 swrast_dri.so segfaults

### ISSUE THREAD

**Issue by mikemorris** [mapbox/mason#51]
Tested `mapbox-gl-native` against [`mesa-10.3.5`](https://github.com/mapbox/mason/tree/mesa-10.3.5) and it looks like there's a segfault coming from `swrast_dri.so` that I haven't seen running against [`mesa-10.3.1`](https://github.com/mapbox/mason/tree/mesa-10.3.1).

```
Traceback (most recent call last):
  File "./../../test/suite/bin/server.py", line 29, in <module>
    server.serve_forever()
  File "/usr/lib/python2.7/SocketServer.py", line 225, in serve_forever
    r, w, e = select.select([self], [], [], poll_interval)
select.error: (4, 'Interrupted system call')
./scripts/run_tests.sh: line 8: 16664 Segmentation fault      (core dumped) ${TEST}
The program crashed with exit code 139. We're now trying to output the core dump.
[New LWP 16664]
[New LWP 16667]
[New LWP 16672]
[New LWP 17472]
[New LWP 17473]
[New LWP 16673]
[New LWP 16666]
[New LWP 16669]
[New LWP 16668]
[New LWP 16678]
[New LWP 17471]
[New LWP 17474]
[New LWP 16670]
[New LWP 16671]

warning: Can't read pathname for load map: Input/output error.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

warning: no loadable sections found in added symbol-file system-supplied DSO at 0x7fff69eb0000
Core was generated by `./test_headless'.
Program terminated with signal 11, Segmentation fault.
#0  0x00007f9aa0077e75 in lp_setup_bin_triangle () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so

Thread 14 (Thread 0x7f9a9a4ea700 (LWP 16671)):
#0  0x00007f9aa8193d84 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f9aa006ea43 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#2  0x00007f9aa818fe9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#3  0x00007f9aa56f24bd in clone () from /lib/x86_64-linux-gnu/libc.so.6
#4  0x0000000000000000 in ?? ()

Thread 13 (Thread 0x7f9a9aeeb700 (LWP 16670)):
#0  0x00007f9aa8193d84 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f9aa006ea43 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#2  0x00007f9aa818fe9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#3  0x00007f9aa56f24bd in clone () from /lib/x86_64-linux-gnu/libc.so.6
#4  0x0000000000000000 in ?? ()

Thread 12 (Thread 0x7f9a986e7700 (LWP 17474)):
#0  0x00007f9aa8193d84 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x000000000050af27 in uv_cond_wait (cond=0x12f8cc8, mutex=0x12f8ca0) at src/unix/thread.c:322
#2  0x00000000004784ab in uv_chan_receive ()
#3  0x0000000000477c59 in uv__worker_thread_loop ()
#4  0x00000000005101a3 in uv__thread_start (ctx_v=0x230be90) at src/uv-common.c:322
#5  0x00007f9aa818fe9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#6  0x00007f9aa56f24bd in clone () from /lib/x86_64-linux-gnu/libc.so.6
#7  0x0000000000000000 in ?? ()

Thread 11 (Thread 0x7f9a972e5700 (LWP 17471)):
#0  0x00007f9aa8193d84 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x000000000050af27 in uv_cond_wait (cond=0x12f8cc8, mutex=0x12f8ca0) at src/unix/thread.c:322
#2  0x00000000004784ab in uv_chan_receive ()
#3  0x0000000000477c59 in uv__worker_thread_loop ()
#4  0x00000000005101a3 in uv__thread_start (ctx_v=0x2b785a0) at src/uv-common.c:322
#5  0x00007f9aa818fe9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#6  0x00007f9aa56f24bd in clone () from /lib/x86_64-linux-gnu/libc.so.6
#7  0x0000000000000000 in ?? ()

Thread 10 (Thread 0x7f9a95de2700 (LWP 16678)):
#0  0x00007f9aa56ee6d9 in syscall () from /lib/x86_64-linux-gnu/libc.so.6
#1  0x00000000005146e2 in uv__epoll_wait (epfd=10, events=0x7f9a95ddecd0, nevents=1024, timeout=-1) at src/unix/linux-syscalls.c:282
#2  0x0000000000511733 in uv__io_poll (loop=0x17d00a0, timeout=-1) at src/unix/linux-core.c:187
#3  0x0000000000501a39 in uv_run (loop=0x17d00a0, mode=UV_RUN_DEFAULT) at src/unix/core.c:317
#4  0x0000000000531b40 in mbgl::thread_init(void*) ()
#5  0x00000000005101a3 in uv__thread_start (ctx_v=0x15d8e90) at src/uv-common.c:322
#6  0x00007f9aa818fe9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#7  0x00007f9aa56f24bd in clone () from /lib/x86_64-linux-gnu/libc.so.6
#8  0x0000000000000000 in ?? ()

Thread 9 (Thread 0x7f9a9c2ed700 (LWP 16668)):
#0  0x00007f9aa8193d84 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f9aa006ea43 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#2  0x00007f9aa818fe9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#3  0x00007f9aa56f24bd in clone () from /lib/x86_64-linux-gnu/libc.so.6
#4  0x0000000000000000 in ?? ()

Thread 8 (Thread 0x7f9a9b8ec700 (LWP 16669)):
#0  0x00007f9aa8193d84 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f9aa006ea43 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#2  0x00007f9aa818fe9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#3  0x00007f9aa56f24bd in clone () from /lib/x86_64-linux-gnu/libc.so.6
#4  0x0000000000000000 in ?? ()

Thread 7 (Thread 0x7f9a9d6ef700 (LWP 16666)):
#0  0x00007f9aa8193d84 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f9aa006ea43 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#2  0x00007f9aa818fe9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#3  0x00007f9aa56f24bd in clone () from /lib/x86_64-linux-gnu/libc.so.6
#4  0x0000000000000000 in ?? ()

Thread 6 (Thread 0x7f9a990e8700 (LWP 16673)):
#0  0x00007f9aa8193d84 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f9aa006ea43 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#2  0x00007f9aa818fe9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#3  0x00007f9aa56f24bd in clone () from /lib/x86_64-linux-gnu/libc.so.6
#4  0x0000000000000000 in ?? ()

Thread 5 (Thread 0x7f9a97ce6700 (LWP 17473)):
#0  0x00007f9aa8193d84 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x000000000050af27 in uv_cond_wait (cond=0x12f8cc8, mutex=0x12f8ca0) at src/unix/thread.c:322
#2  0x00000000004784ab in uv_chan_receive ()
#3  0x0000000000477c59 in uv__worker_thread_loop ()
#4  0x00000000005101a3 in uv__thread_start (ctx_v=0x2edbc10) at src/uv-common.c:322
#5  0x00007f9aa818fe9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#6  0x00007f9aa56f24bd in clone () from /lib/x86_64-linux-gnu/libc.so.6
#7  0x0000000000000000 in ?? ()

Thread 4 (Thread 0x7f9a968e4700 (LWP 17472)):
#0  0x00007f9aa8193d84 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x000000000050af27 in uv_cond_wait (cond=0x12f8cc8, mutex=0x12f8ca0) at src/unix/thread.c:322
#2  0x00000000004784ab in uv_chan_receive ()
#3  0x0000000000477c59 in uv__worker_thread_loop ()
#4  0x00000000005101a3 in uv__thread_start (ctx_v=0x12fe840) at src/uv-common.c:322
#5  0x00007f9aa818fe9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#6  0x00007f9aa56f24bd in clone () from /lib/x86_64-linux-gnu/libc.so.6
#7  0x0000000000000000 in ?? ()

Thread 3 (Thread 0x7f9a99ae9700 (LWP 16672)):
#0  0x00007f9aa8193d84 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f9aa006ea43 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#2  0x00007f9aa818fe9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#3  0x00007f9aa56f24bd in clone () from /lib/x86_64-linux-gnu/libc.so.6
#4  0x0000000000000000 in ?? ()

Thread 2 (Thread 0x7f9a9ccee700 (LWP 16667)):
#0  0x00007f9aa8193d84 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f9aa006ea43 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#2  0x00007f9aa818fe9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#3  0x00007f9aa56f24bd in clone () from /lib/x86_64-linux-gnu/libc.so.6
#4  0x0000000000000000 in ?? ()

Thread 1 (Thread 0x7f9aa08a2780 (LWP 16664)):
#0  0x00007f9aa0077e75 in lp_setup_bin_triangle () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#1  0x00007f9aa00789d0 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#2  0x00007f9aa0078b61 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#3  0x00007f9aa0079785 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#4  0x00007f9aa008e591 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#5  0x00007f9aa008ec02 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#6  0x00007f9aa00853d7 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#7  0x00007f9aa0086f67 in draw_pipeline_run () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#8  0x00007f9aa0113fa5 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#9  0x00007f9aa0114103 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#10 0x00007f9aa0096387 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#11 0x00007f9aa008fc27 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#12 0x00007f9aa008ff66 in draw_vbo () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#13 0x00007f9aa006dc41 in ?? () from /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so
#14 0x00007f9a9f4cd84b in st_draw_vbo () from /usr/lib/x86_64-linux-gnu/dri/libgallium.so
#15 0x00007f9a9fca6a57 in ?? () from /usr/lib/x86_64-linux-gnu/dri/libdricore.so
#16 0x00007f9a9fca6dac in ?? () from /usr/lib/x86_64-linux-gnu/dri/libdricore.so
#17 0x00000000004d4720 in mbgl::FillBucket::drawElements(mbgl::PlainShader&) ()
#18 0x00000000005008f7 in mbgl::Painter::renderFill(mbgl::FillBucket&, mbgl::util::ptr<mbgl::StyleLayer>, mbgl::Tile::ID const&, std::array<float, 16ul> const&) ()
#19 0x00000000004d44f8 in mbgl::FillBucket::render(mbgl::Painter&, mbgl::util::ptr<mbgl::StyleLayer>, mbgl::Tile::ID const&, std::array<float, 16ul> const&) ()
#20 0x00000000004b9fa8 in mbgl::VectorTileData::render(mbgl::Painter&, mbgl::util::ptr<mbgl::StyleLayer>, std::array<float, 16ul> const&) ()
#21 0x000000000048714f in mbgl::Painter::renderTileLayer(mbgl::Tile const&, mbgl::util::ptr<mbgl::StyleLayer>, std::array<float, 16ul> const&) ()
#22 0x000000000048f37d in mbgl::Source::render(mbgl::Painter&, mbgl::util::ptr<mbgl::StyleLayer>) ()
#23 0x0000000000486849 in mbgl::Painter::renderLayer(mbgl::util::ptr<mbgl::StyleLayer>, mbgl::Tile::ID const*, std::array<float, 16ul> const*) ()
#24 0x0000000000485cba in mbgl::Painter::renderLayers(mbgl::util::ptr<mbgl::StyleLayerGroup>) ()
#25 0x0000000000485498 in mbgl::Painter::render(mbgl::Style const&, std::set<mbgl::util::ptr<mbgl::StyleSource>, std::less<mbgl::util::ptr<mbgl::StyleSource> >, std::allocator<mbgl::util::ptr<mbgl::StyleSource> > > const&, mbgl::TransformState, unsigned long) ()
#26 0x00000000004701d6 in mbgl::Map::render() ()
#27 0x000000000046fc70 in mbgl::Map::run() ()
#28 0x000000000041de85 in HeadlessTest_render_Test::TestBody() ()
#29 0x000000000045d103 in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) ()
#30 0x000000000044c951 in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) ()
#31 0x0000000000432327 in testing::Test::Run() ()
#32 0x000000000043317c in testing::TestInfo::Run() ()
#33 0x00000000004339a5 in testing::TestCase::Run() ()
#34 0x000000000043ba2b in testing::internal::UnitTestImpl::RunAllTests() ()
#35 0x0000000000459663 in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) ()
#36 0x000000000044f041 in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) ()
#37 0x000000000043b6d7 in testing::UnitTest::Run() ()
#38 0x000000000041a990 in main ()
```

https://travis-ci.org/mapbox/mapbox-gl-native/jobs/47706355

**Comment by mikemorris** [comment]
Curious if the recent bug-fix releases [10.3.6](http://mesa3d.org/relnotes/10.3.6.html) or [10.3.7](http://mesa3d.org/relnotes/10.3.7.html) fixed this.

> NOTE: It is anticipated that 10.3.7 will be the final release in the 10.3 series. Users of 10.3 are encouraged to migrate to the 10.4 series in order to obtain future fixes.
http://mesa3d.org/

Probably worth looking into building [Mesa 10.4.2](http://mesa3d.org/relnotes/10.4.2.html).

**Comment by jfirebaugh** [comment]
Stale.
